diff --git a/CHANGELOG b/CHANGELOG
new file mode 100644
index 0000000..f35aa9b
--- /dev/null
+++ b/CHANGELOG
@@ -0,0 +1,14 @@
+0.6c
+DEBUG mode constant is defined in debug mode (check for DEBUG being set)
+Bootstrap introduced to set up logging and debug mode
+Bug fix: Crash with error on failed DB
+
+0.6b
+Security fix: Don't display peoples passwords in their profiles
+Anti-caching technology enabled
+
+0.6a
+Some bugs were introduced
+
+0.6
+Lost the tides of time
diff --git a/VERSION b/VERSION
new file mode 100644
index 0000000..58770f6
--- /dev/null
+++ b/VERSION
@@ -0,0 +1 @@
+0.6c
diff --git a/bootstrap.php b/bootstrap.php
new file mode 100644
index 0000000..bd4135c
--- /dev/null
+++ b/bootstrap.php
@@ -0,0 +1,18 @@
+<?php
+
+//Set up logging
+ini_set("log_errors", 1);
+ini_set("error_log", "error.log");
+
+//Check for database
+if(!$f3->get('db')) {
+	die('Unable to read database configuration. Ensure your database configuration exists and is correct');
+}
+
+//Check for settings 
+$settings = Settings::getSettings();
+if($settings['debug'] == 1) {
+
+	//Define DEBUG mode as 1 if debug mode is enabled
+	define('DEBUG',1);
+}
diff --git a/controllers/admin/settings.php b/controllers/admin/settings.php
index 3f4fcd2..ec52808 100644
--- a/controllers/admin/settings.php
+++ b/controllers/admin/settings.php
@@ -1,26 +1,41 @@
 <?php
 
-	namespace Admin;
-
-	class Settings extends AdminController {
-
-		public function index($f3) {
-			$settings = $this->Model->Settings->fetchAll();
-			if($this->request->is('post')) {
-				foreach($settings as $setting) {
-					if(isset($this->request->data[$setting->setting])) {
-						$setting->value = $this->request->data[$setting->setting];
-						$setting->save();
-					} else {
-						$setting->value = 0;
-						$setting->save();
-					}
+namespace Admin;
+
+class Settings extends AdminController {
+
+	public function index($f3) {
+		$settings = $this->Model->Settings->fetchAll();
+		if($this->request->is('post')) {
+			foreach($settings as $setting) {
+				if(isset($this->request->data[$setting->key])) {
+					$setting->value = $this->request->data[$setting->key];
+					$setting->save();
+				} else {
+					$setting->value = 0;
+					$setting->save();
 				}
-				\StatusMessage::add('Settings updated','success');
 			}
-			$f3->set('settings',$settings);
+			\StatusMessage::add('Settings updated','success');
 		}
+		$f3->set('settings',$settings);
+	}
 
+	public function clearcache($f3) {
+		$cache = isset($this->request->data['cache']) ? getcwd() . '/' . $this->request->data['cache'] : getcwd() . '/tmp/cache';
+		$cache = str_replace(".","",$cache);
+		$this->delTree($cache);
 	}
 
+	public function delTree($dir) { 
+		$files = array_diff(scandir($dir), array('.','..')); 
+		foreach ($files as $file) {
+			(is_dir("$dir/$file") && !is_link($dir)) ? $this->delTree("$dir/$file") : unlink("$dir/$file"); 
+		}
+		return rmdir($dir); 
+	} 
+
+}
+
+
 ?>
diff --git a/controllers/controller.php b/controllers/controller.php
index 01f6523..903bfdc 100644
--- a/controllers/controller.php
+++ b/controllers/controller.php
@@ -27,7 +27,7 @@ class Controller {
 		$this->Auth->resume();
 
 		//Load settings
-		$settings = $this->Model->Settings->fetchList(array('setting','value'));
+		$settings = Settings::getSettings();
 		$settings['base'] = $f3->get('BASE');
 		
 		//Append debug mode to title
@@ -37,7 +37,7 @@ class Controller {
 		$this->Settings = $settings;
 		$f3->set('site',$settings);
 
-				//Extract request data
+		//Extract request data
 		extract($this->request->data);
 
 		//Process before route code
diff --git a/controllers/user.php b/controllers/user.php
index 4c022ed..61673f9 100644
--- a/controllers/user.php
+++ b/controllers/user.php
@@ -87,8 +87,10 @@ class User extends Controller {
 		$id = $this->Auth->user('id');
 		extract($this->request->data);
 		$u = $this->Model->Users->fetch($id);
+		$oldpass = $u->password;
 		if($this->request->is('post')) {
 			$u->copyfrom('POST');
+			if(empty($u->password)) { $u->password = $oldpass; }
 
 			//Handle avatar upload
 			if(isset($_FILES['avatar']) && isset($_FILES['avatar']['tmp_name']) && !empty($_FILES['avatar']['tmp_name'])) {
diff --git a/index.php b/index.php
index 2a9f97e..67d3e55 100644
--- a/index.php
+++ b/index.php
@@ -1,9 +1,5 @@
 <?php
 
-//Set up logging
-ini_set("log_errors", 1);
-ini_set("error_log", "error.log");
-
 //Load framework
 $f3=require('lib/base.php');
 $f3->config('config/config.cfg');
@@ -13,6 +9,7 @@ $f3->set('AUTOLOAD','controllers/; models/; helpers/; utility/;');
 $f3->config('config/db.cfg');
 
 //Load global functions
+include_once("bootstrap.php");
 include_once("functions.php");
 
 //Define homepage 
diff --git a/models/database.php b/models/database.php
index b98bb43..4c97463 100644
--- a/models/database.php
+++ b/models/database.php
@@ -2,18 +2,26 @@
 
 class Database {
 	
+	public static $singleton;
 	public $connection;
+	
+	/** Return the single database connection */
+	public static function getConnection() {
+		if(empty(self::$singleton)) {
+			$f3=Base::instance();
+			extract($f3->get('db'));
+			self::$singleton=new DB\SQL(
+					'mysql:host='.$server.';port=3306;dbname='.$name,
+					$username,
+					$password
+					);
+		}
+		return self::$singleton;
+	}	
 
 	/** Create a new database object */
 	public function __construct() {
-		$f3=Base::instance();
-		
-		extract($f3->get('db'));
-		$this->connection=new DB\SQL(
-		    'mysql:host='.$server.';port=3306;dbname='.$name,
-		    $username,
-		    $password
-		);
+		$this->connection = self::getConnection();
 	}
 
 	/** Perform a direct database query */
diff --git a/utility/settings.php b/utility/settings.php
index 1b1b10b..d6266e3 100644
--- a/utility/settings.php
+++ b/utility/settings.php
@@ -2,11 +2,13 @@
 
 class Settings {
 
-	public function __construct() {
+	/** Get all the settings */
+	public static function getSettings() {		
+		$model = new SettingsModel('Settings',new Database());
+		$settings = $model->fetchList(array('setting','value'));
+		return $settings;
 	}
 
-
-
 }
 
 ?>
diff --git a/view/Admin/admin_index.htm b/view/Admin/admin_index.htm
index 48f7214..4628447 100644
--- a/view/Admin/admin_index.htm
+++ b/view/Admin/admin_index.htm
@@ -23,7 +23,7 @@
 <div class="well">
 <h3>Daily Rob</h3>
 <div style="text-align: center">
-<img src="http://linuxproj.ecs.soton.ac.uk/~ofb1v07/rob.jpg" style="max-height: 150px; width: auto;"/>
+<img src="http://robpress.clicked.cc/rob.jpg" style="max-height: 150px; width: auto;"/>
 </div>
 </div>
 </div>
diff --git a/view/User/profile.htm b/view/User/profile.htm
index ea21bbd..299133c 100644
--- a/view/User/profile.htm
+++ b/view/User/profile.htm
@@ -10,7 +10,7 @@
 	$form = new Form();
 	echo $form->start(array('type' => 'file'));
 	echo $form->add('displayname',array('label' => 'Display Name'));
-	echo $form->add('password',array('type' => 'password'));
+	echo $form->add('password',array('type' => 'password', 'label' => 'Change Password','value' => ''));
 	echo $form->add('bio',array('type' => 'wysiwyg'));
 	echo $form->add('avatar',array('type' => 'file'));
 	if(!empty($u->avatar)) {
diff --git a/view/header.htm b/view/header.htm
index 165c045..33227a9 100644
--- a/view/header.htm
+++ b/view/header.htm
@@ -5,7 +5,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <meta name="description" content="">
-    <meta name="generator" content="RobPress 0.6a - http://robpress.clicked.cc">
+    <meta name="generator" content="RobPress 0.6c - http://robpress.clicked.cc">
     <?if($site['debug']):?>
     <meta name="debug" content="Debug mode is enabled"/>
     <?endif?>
-- 
1.9.1

